### ISSUE TITLE
Refresh data table without refreshing whole site/component

### ISSUE THREAD

**Issue by kalwinskidawid** [Gustavinho/laravel-views#164]
How to refresh data / repository in tableview without refreshing whole site?

**Comment by Gustavinho** [comment]
Hi @kalwinskidawid, there's no "Refresh" button, however, every filter or sort action is refreshing the data, thinking this through, a refresh button would be great

**Comment by kalwinskidawid** [comment]
Okay, tho

**Comment by kalwinskidawid** [comment]
So should I use `$this->render()` after change any data in database?

**Comment by Gustavinho** [comment]
yes `$this->render` should work, although it wasn't the intention to have to update the data manually, do you have something like a form or something else updating the data outside of the table in the same page?

**Comment by kalwinskidawid** [comment]
Yes!  I have a modal from WireUI which I use to dynamically edit and add data without page refresh. After validating the form, it sends a signal to the TableView to reload data in the table

**Comment by Gustavinho** [comment]
In that case, I'll suggest using a livewire event, probably you're already using it, and the `render()` method of the table as you said, that should work

**Comment by christoferd** [comment]
Emit a [Livewire Event](https://laravel-livewire.com/docs/2.x/events)
Catch it in your Livewire Component then updating the records.
I have a separate function to get the list of data. This function will update $this->events that is called when the event is caught.
This is the only way I could think of having an add event component open in a modal, do its work, then the calendar update ONLY the events after adding an event to the database.

**Comment by christoferd** [comment]
I've been thinking about this lately, and I guess the best way is to separate the table into smaller components. 
So that **each row is one component**.
That way the "Record Updated" event may be caught by the table if required, but more importantly - by each row component.
...
Then obviously, the row component would check to see if the updated record matches the ID of that row, and if so, it would update the record within that component's class... so Livewire sends just a bit of data.

**Comment by Gustavinho** [comment]
Hi @christoferd I'll play around with this approach using individual components for each row, this will also be helpful for another concern about the size of the data on every update

**Comment by christoferd** [comment]
Also, best practice not to send/share models with components. If data is all the component needs, then a Collection, json string or array should be sent.
